client/my-sites/checkout/composite-checkout/lib/leave-checkout.ts

```diff
@@ -11,15 +11,17 @@ export const leaveCheckout = ( {
 	jetpackCheckoutBackUrl,
 	previousPath,
 	dispatch,
+	tracksEvent,
 	createUserAndSiteBeforeTransaction,
 }: {
 	siteSlug?: string;
 	jetpackCheckoutBackUrl?: string;
 	previousPath?: string;
 	dispatch: ReturnType< typeof useDispatch >;
+	tracksEvent: string;
 	createUserAndSiteBeforeTransaction?: boolean;
 } ): void => {
-	dispatch( recordTracksEvent( 'calypso_masterbar_close_clicked' ) );
+	dispatch( recordTracksEvent( tracksEvent ) );
```

Comment:
A little framework suggestion for `leaveCheckout`. Either rewrite it to a regular Redux action thunk:
```js
const leaveCheckout = ( { siteSlug, ... } ) => ( dispatch ) => {
   ...
}
```
and use it like:
```js
reduxDispatch( leaveCheckout( { siteSlug, ... } ) );
```
Or remove the need for `dispatch` completely, by calling the `recordTracksEvent` _function_ from `calypso/lib/analytics/tracks` instead of dispatching the _action_ from `calypso/state/analytics/actions`.